Cesium3DTilesSelection/src/Tile.cpp

```diff
@@ -306,74 +337,80 @@ void Tile::loadContent() {
             loadInput.data = pResponse->data();
             loadInput.contentType = pResponse->contentType();
             loadInput.url = pRequest->url();
-            std::unique_ptr<TileContentLoadResult> pContent =
-                TileContentFactory::createContent(loadInput);
-
-            void* pRendererResources = nullptr;
-            if (pContent) {
-              pContent->httpStatusCode = pResponse->statusCode();
-
-              if (pContent->model) {
-
-                CesiumGltf::Model& model = pContent->model.value();
-
-                // TODO The `extras` are currently the only way to pass
-                // arbitrary information to the consumer, so the up-axis
-                // is stored here:
-                model.extras["gltfUpAxis"] = gltfUpAxis;
-
-                const BoundingVolume& boundingVolume =
-                    loadInput.tileBoundingVolume;
-                // TODO: apply gltf up axis (if it's not the glTF standard Y-up,
-                // which is already accounted for) and RTC_CENTER.
-                const glm::dmat4& transform = loadInput.tileTransform;
-
-                Tile::generateTextureCoordinates(
-                    model,
-                    transform,
-                    boundingVolume,
-                    projections);
-
-                pContent->rasterOverlayProjections = std::move(projections);
-
-                if (generateMissingNormalsSmooth) {
-                  pContent->model->generateMissingNormalsSmooth();
-                }
-
-                if (pPrepareRendererResources) {
-                  CESIUM_TRACE("prepareInLoadThread");
-                  pRendererResources =
-                      pPrepareRendererResources->prepareInLoadThread(
-                          pContent->model.value(),
-                          transform);
-                }
-              }
-            }
-
-            LoadResult result;
-            result.state = LoadState::ContentLoaded;
-            result.pContent = std::move(pContent);
-            result.pRendererResources = pRendererResources;
 
-            return result;
+            TileContentFactory::createContent(
```

Comment:
You should return the Future returned by createContent. That way the outer Future doesn't resolve until the one created by createContent does. I think that will greatly simplify this method (i.e. you don't need the two levels of handling result/errors).

Reply:
Wouldn't that end up becoming a "future of a future"? Or will that automatically be unwrapped twice in the result-handler?

Reply:
It'll get unwrapped. See https://github.com/Amanieu/asyncplusplus/wiki/Tasks#task-unwraping and replace "task" with "future". :)